blueocean-dashboard/src/main/js/components/records.jsx

```diff
@@ -65,6 +69,15 @@ export const RunsRecord = Record({
     pullRequest: PullRequestRecord,
 });
 
+export class FolderRecord extends Record({
```

Comment:
Still not sure if a separate folder record is the right way. I can see what you have done, and it makes sense, but will we have records of all types at some point and then the listing is always heterogenous? (not sure why, but heterogenous lists seem... odd... although I know this is what the json returns). I suppose it is better than a generic object with half the things null. 

My kingdom for sum types ;)